addons/payment_demo/static/src/js/express_checkout_form.js

```diff
@@ -23,6 +23,34 @@ paymentExpressCheckoutForm.include({
         this._initiateExpressPayment = debounce(this._initiateExpressPayment, 500, true);
     },
 
+    /**
+     * Prepare the express checkout form of Stripe for direct payment.
+     *
+     * @override method from payment.express_form
+     * @private
+     * @param {Object} providerData - The provider-specific data.
+     * @return {void}
+     */
+    async _prepareExpressCheckoutForm(providerData) {
+        /*
+         * When applying a coupon, the amount can be totally covered, with nothing left to pay. In
+         * that case, the check is whether the variable is defined because the server doesn't send
+         * the value when it equals '0'.
+         */
+        if (providerData.providerCode !== 'demo' || !this.paymentContext['amount']) {
+            this._super(...arguments);
+            return;
```

Comment:
?
```suggestion
            return this._super(...arguments);
```

Reply:
I just tried to be consistent with the new code in payment. Since the method now returns `void` if feels weird to return `super`

Reply:
Well, with my suggestion, you return what the parent method returns, whatever it is. With your way, you'd need to adapt if the parent signature changes. 

But, again, I'm not used to the payment JS methods, so feel free to discard.